storage-server-api/src/main/java/org/jetbrains/ruby/codeInsight/types/storage/server/impl/Schema.kt

```diff
@@ -4,15 +4,91 @@ import org.jetbrains.exposed.dao.EntityID
 import org.jetbrains.exposed.dao.IntEntity
 import org.jetbrains.exposed.dao.IntEntityClass
 import org.jetbrains.exposed.dao.IntIdTable
-import org.jetbrains.exposed.sql.ReferenceOption
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.like
+import org.jetbrains.exposed.sql.statements.InsertStatement
+import org.jetbrains.exposed.sql.statements.UpdateBuilder
+import org.jetbrains.exposed.sql.statements.UpdateStatement
+import org.jetbrains.exposed.sql.transactions.TransactionManager
+import org.jetbrains.exposed.sql.transactions.transaction
 import org.jetbrains.ruby.codeInsight.types.signature.*
 import org.jetbrains.ruby.codeInsight.types.signature.serialization.BlobDeserializer
+import org.jetbrains.ruby.codeInsight.types.signature.serialization.BlobSerializer
 import java.sql.Blob
 import kotlin.reflect.KProperty
 
-object GemInfoTable : IntIdTable() {
+abstract class IntIdTableWithPossibleDependency<in T, out F>(
```

Comment:
This is a nice approach for chaining. However, it should not be placed along with schema because it complicated schema reading much. Please extract it into a separate file. I am not sure if this abstract class belongs to Table type hierarchy (I'd say it's a separate service), but it surely does not belong to schema file.